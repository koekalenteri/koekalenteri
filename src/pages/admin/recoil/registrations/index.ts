export * from './atoms'
// 2nd file to avoid circular deps
export * from './atoms2'
export * from './selectors'
