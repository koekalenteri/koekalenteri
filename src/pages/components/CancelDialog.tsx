import type { SelectChangeEvent } from '@mui/material/Select'
import type { ChangeEvent } from 'react'
import type { PublicDogEvent, Registration } from '../../types'

import { useCallback, useState } from 'react'
import { Trans, useTranslation } from 'react-i18next'
import Button from '@mui/material/Button'
import Dialog from '@mui/material/Dialog'
import DialogActions from '@mui/material/DialogActions'
import DialogContent from '@mui/material/DialogContent'
import DialogContentText from '@mui/material/DialogContentText'
import DialogTitle from '@mui/material/DialogTitle'
import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import InputLabel from '@mui/material/InputLabel'
import Link from '@mui/material/Link'
import MenuItem from '@mui/material/MenuItem'
import Select from '@mui/material/Select'
import TextField from '@mui/material/TextField'

import { isPredefinedReason } from '../../lib/registration'

interface Props {
  readonly admin?: boolean
  readonly disabled?: boolean
  readonly event: PublicDogEvent | null | undefined
  readonly onCancel?: (reason: string) => void
  readonly onClose?: () => void
  readonly open: boolean
  readonly registration: Registration | null | undefined
}

const CancelDialog = ({ admin, disabled, event, onCancel, onClose, open, registration }: Props) => {
  const { t } = useTranslation()
  const [reason, setReason] = useState(
    registration?.cancelReason && isPredefinedReason(registration.cancelReason) ? registration.cancelReason : ''
  )
  const [freeReason, setFreeReason] = useState(
    registration?.cancelReason && !isPredefinedReason(registration.cancelReason) ? registration?.cancelReason : ''
  )

  const handleReasonChange = useCallback((event: SelectChangeEvent) => {
    setReason(event.target.value)
  }, [])

  const handleFreeReasonChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
    setFreeReason(event.target.value)
  }, [])

  const handleCancel = useCallback(
    () => onCancel?.(reason === 'other' ? freeReason : reason),
    [freeReason, onCancel, reason]
  )

  if (!event || !registration) {
    return null
  }

  return (
    <Dialog
      open={open}
      onClose={onClose}
      aria-labelledby="cancel-dialog-title"
      aria-describedby="cancel-dialog-description"
    >
      <DialogTitle id="cancel-dialog-title">{t('registration.cancelDialog.title')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="cancel-dialog-description">
          {disabled
            ? t('registration.cancelDialog.lateText', {
                registration,
                event,
                contact: event.contactInfo?.secretary?.phone ?? event.contactInfo?.secretary?.email,
              })
            : t('registration.cancelDialog.text', { registration, event })}
        </DialogContentText>
        <DialogContentText id="cancel-dialog-description2" sx={{ py: 1, display: disabled ? 'none' : 'block' }}>
          {t('registration.cancelDialog.confirmation')}
        </DialogContentText>
        <FormControl fullWidth sx={{ my: 1 }}>
          <InputLabel id="reason-label">{t('registration.cancelDialog.reason')}</InputLabel>
          <Select
            id="reaseon"
            labelId="reason-label"
            label={t('registration.cancelDialog.reason')}
            value={reason}
            onChange={handleReasonChange}
          >
            <MenuItem value="dog-heat">{t('registration.cancelReason.dog-heat')}</MenuItem>
            <MenuItem value="handler-sick">{t('registration.cancelReason.handler-sick')}</MenuItem>
            <MenuItem value="dog-sick">{t('registration.cancelReason.dog-sick')}</MenuItem>
            <MenuItem value="other">{t('registration.cancelReason.other')}</MenuItem>
            <MenuItem value="gdpr">{t('registration.cancelReason.gdpr')}</MenuItem>
          </Select>
          <FormHelperText>
            {!admin && (reason === 'handler-sick' || reason === 'dog-sick') && (
              <>
                {t(`registration.cancelReason.${reason}-info`)} &nbsp;
                <a href={`mailto://${event.contactInfo?.secretary?.email}`}>{event.contactInfo?.secretary?.email}</a>
              </>
            )}
          </FormHelperText>
        </FormControl>
        {reason === 'other' && (
          <TextField
            fullWidth
            label={t('registration.cancelDialog.reason_other')}
            value={freeReason}
            onChange={handleFreeReasonChange}
          />
        )}
        <DialogContentText id="cancel-dialog-description3" sx={{ py: 1 }}>
          <Trans t={t} i18nKey="registration.cancelDialog.terms">
            Katso tarkemmat peruutusehdot{' '}
            <Link
              target="_blank"
              rel="noopener"
              href="https://yttmk.yhdistysavain.fi/noutajien-metsastyskokeet-2/ohjeistukset/kokeen-ja-tai-kilpailun-ilmoitta/"
            >
              säännöistä
            </Link>
            .
          </Trans>
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={handleCancel}
          disabled={disabled || reason === '' || (reason === 'other' && freeReason === '')}
          autoFocus
          variant="contained"
        >
          {t('registration.cancelDialog.cta')}
        </Button>
        <Button onClick={onClose} variant="outlined">
          {t('close')}
        </Button>
      </DialogActions>
    </Dialog>
  )
}

export default CancelDialog
