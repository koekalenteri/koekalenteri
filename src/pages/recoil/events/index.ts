export * from './atoms'
export * from './effects'
export * from './filters'
export * from './selectors'
export * from './types'
