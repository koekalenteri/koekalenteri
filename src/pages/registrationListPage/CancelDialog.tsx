import type { TFunction } from 'i18next'
import type { PublicConfirmedEvent, Registration } from '../../types'

import { Trans } from 'react-i18next'
import Button from '@mui/material/Button'
import Dialog from '@mui/material/Dialog'
import DialogActions from '@mui/material/DialogActions'
import DialogContent from '@mui/material/DialogContent'
import DialogContentText from '@mui/material/DialogContentText'
import DialogTitle from '@mui/material/DialogTitle'
import Link from '@mui/material/Link'

interface Props {
  readonly disabled?: boolean
  readonly event: PublicConfirmedEvent | null | undefined
  readonly onCancel?: () => void
  readonly onClose?: () => void
  readonly open: boolean
  readonly registration: Registration | null | undefined
  readonly t: TFunction
}

export function CancelDialog({ disabled, event, onCancel, onClose, open, registration, t }: Props) {
  if (!event || !registration) {
    return null
  }

  return (
    <Dialog
      open={open}
      onClose={onClose}
      aria-labelledby="cancel-dialog-title"
      aria-describedby="cancel-dialog-description"
    >
      <DialogTitle id="cancel-dialog-title">{t('registration.cancelDialog.title')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="cancel-dialog-description">
          {disabled
            ? t('registration.cancelDialog.lateText', {
                registration,
                event,
                contact: event.contactInfo?.secretary?.phone ?? event.contactInfo?.secretary?.email,
              })
            : t('registration.cancelDialog.text', { registration, event })}
        </DialogContentText>
        <DialogContentText id="cancel-dialog-description2" sx={{ py: 1, display: disabled ? 'none' : 'block' }}>
          {t('registration.cancelDialog.confirmation')}
        </DialogContentText>
        <DialogContentText id="cancel-dialog-description3" sx={{ py: 1 }}>
          <Trans t={t} i18nKey="registration.cancelDialog.terms">
            Katso tarkemmat peruutusehdot{' '}
            <Link
              target="_blank"
              rel="noopener"
              href="https://yttmk.yhdistysavain.fi/noutajien-metsastyskokeet-2/ohjeistukset/kokeen-ja-tai-kilpailun-ilmoitta/"
            >
              säännöistä
            </Link>
            .
          </Trans>
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onCancel} disabled={disabled} autoFocus variant="contained">
          {t('registration.cancelDialog.cta')}
        </Button>
        <Button onClick={onClose} variant="outlined">
          {t('cancel')}
        </Button>
      </DialogActions>
    </Dialog>
  )
}
