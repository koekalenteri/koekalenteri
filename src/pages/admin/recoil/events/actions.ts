import type { DogEvent } from '../../../../types'

import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import { addDays, differenceInDays } from 'date-fns'
import { diff } from 'deep-object-diff'
import { useSnackbar } from 'notistack'
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil'

import { copyEventWithRegistrations, putEvent } from '../../../../api/event'
import { sanitizeDogEvent } from '../../../../lib/event'
import { Path } from '../../../../routeConfig'
import { eventsAtom, idTokenAtom, userSelector } from '../../../recoil'

import {
  adminEventIdAtom,
  adminNewEventAtom,
  newEventEntryEndDate,
  newEventEntryStartDate,
  newEventStartDate,
} from './atoms'
import { adminCurrentEventSelector } from './selectors'

export const useAdminEventActions = () => {
  const token = useRecoilValue(idTokenAtom)
  const user = useRecoilValue(userSelector)
  const setAdminEventId = useSetRecoilState(adminEventIdAtom)
  const [currentAdminEvent, setCurrentAdminEvent] = useRecoilState(adminCurrentEventSelector)
  const setNewEvent = useSetRecoilState(adminNewEventAtom)
  const [publicEvents, setPublicEvents] = useRecoilState(eventsAtom)
  const { enqueueSnackbar } = useSnackbar()
  const { t } = useTranslation()
  const navigate = useNavigate()

  return {
    copyCurrent,
    copyCurrentTest,
    deleteCurrent,
    save,
  }

  function updatePublicEvents(event: DogEvent, remove?: boolean): void {
    if (event.id) {
      const index = publicEvents.findIndex((e) => e.id === event.id)
      if (index >= 0) {
        const newEvents = [...publicEvents]
        if (remove) {
          newEvents.splice(index, 1)
        } else {
          newEvents.splice(index, 1, sanitizeDogEvent(event))
        }
        setPublicEvents(newEvents)
      } else if (!remove) {
        setPublicEvents([...publicEvents, sanitizeDogEvent(event)])
      }
    }
  }

  function copyCurrent() {
    if (!currentAdminEvent) {
      return
    }
    const copy = structuredClone(currentAdminEvent)
    copy.id = ''
    copy.name = 'Kopio - ' + (copy.name ?? '')
    copy.state = 'draft'
    copy.entries = 0
    copy.classes.forEach((c) => {
      c.entries = c.members = 0
      if (c.date) {
        c.date = addDays(newEventStartDate, differenceInDays(copy.startDate, c.date))
      }
      delete c.state
    })

    const days = differenceInDays(copy.endDate, copy.startDate)
    copy.startDate = newEventStartDate
    copy.endDate = addDays(newEventStartDate, days)
    copy.entryStartDate = newEventEntryStartDate
    copy.entryEndDate = newEventEntryEndDate

    delete copy.kcId
    delete copy.entryOrigEndDate

    setNewEvent(copy)
    navigate(Path.admin.newEvent)
  }

  async function copyCurrentTest() {
    if (!currentAdminEvent) {
      return
    }
    const saved = await copyEventWithRegistrations(currentAdminEvent.id, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)
    return saved
  }

  async function save(event: Partial<DogEvent>): Promise<DogEvent | undefined> {
    let changes: Partial<DogEvent> = event
    if (event.id && event.id === currentAdminEvent?.id) {
      const changedKeys = Object.keys(diff(currentAdminEvent, event))
      changes = { id: event.id }
      for (const key of changedKeys) {
        //@ts-expect-error typescript is just so stupid sometimes
        changes[key] = event[key]
      }
    }
    console.log(changes)
    const saved = await putEvent(changes, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)

    return saved
  }

  async function deleteCurrent() {
    if (!currentAdminEvent || currentAdminEvent.deletedAt) {
      return
    }

    await save({
      ...currentAdminEvent,
      deletedAt: new Date(),
      deletedBy: user?.name ?? user?.email,
    })

    updatePublicEvents(currentAdminEvent, true)

    enqueueSnackbar(t('deleteEventComplete'), { variant: 'info' })
  }
}
