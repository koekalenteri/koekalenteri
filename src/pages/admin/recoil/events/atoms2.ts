import type { DogEvent, RegistrationClass } from '../../../../types'

import { atom, atomFamily, selector } from 'recoil'

import { uniqueClasses } from '../../../../lib/utils'
import { logEffect, sessionStorageEffect } from '../../../recoil'

import { adminCurrentEventSelector, adminEventSelector } from './selectors'

export const adminEventClassAtom = atom<RegistrationClass | string>({
  key: 'adminEventClass',
  default: selector({
    key: 'adminEventClass/default',
    get: ({ get }) => {
      const event = get(adminCurrentEventSelector)
      return uniqueClasses(event)[0]
    },
  }),
  effects: [logEffect, sessionStorageEffect],
})

/**
 * Existing event editing, edits stored to session storage
 */
export const adminEditableEventByIdAtom = atomFamily<DogEvent | undefined, string>({
  key: 'adminEditableEvent/Id',
  default: adminEventSelector,
  effects: [logEffect, sessionStorageEffect],
})
