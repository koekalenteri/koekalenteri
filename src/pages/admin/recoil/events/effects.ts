import type { AtomEffect } from 'recoil'
import type { DogEvent } from '../../../../types'

import { DefaultValue } from 'recoil'

import { getAdminEvents } from '../../../../api/event'
import { idTokenAtom } from '../../../recoil'

const byStartDate = (a: DogEvent, b: DogEvent) => a.startDate.valueOf() - b.startDate.valueOf()
const sortEvents = (events: DogEvent[]): DogEvent[] => [...events].sort(byStartDate)

export const adminRemoteEventsEffect: AtomEffect<DogEvent[]> = ({ getPromise, setSelf, trigger }) => {
  if (trigger === 'get') {
    setSelf(
      getPromise(idTokenAtom).then((token) =>
        token ? getAdminEvents(token).then((events) => sortEvents(events)) : new DefaultValue()
      )
      /*
      Promise.all([getPromise(idTokenAtom), getPromise(adminEventsFetchedAtom)]).then(([token, lastModified]) =>
        token ? getAdminEvents(token, lastModified).then((events) => sortEvents(events)) : new DefaultValue()
      )
      */
    )
  }
}
