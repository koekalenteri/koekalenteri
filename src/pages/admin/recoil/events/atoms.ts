import type { GridColumnVisibilityModel } from '@mui/x-data-grid'
import type { DogEvent, RegistrationClass } from '../../../../types'

import { addDays, addYears, isSameDay, nextSaturday, sub } from 'date-fns'
import { atom, atomFamily, selector } from 'recoil'

import { zonedStartOfDay } from '../../../../i18n/dates'
import { uniqueClasses } from '../../../../lib/utils'
import { localStorageEffect, logEffect, sessionStorageEffect } from '../../../recoil'

import { adminRemoteEventsEffect } from './effects'
import { adminCurrentEventSelector, adminEventSelector } from './selectors'

export const adminEventsFetchedAtom = atom<number | undefined>({
  key: 'adminEventsFetched',
  default: addYears(new Date(), -1).valueOf(),
  effects: [logEffect, sessionStorageEffect],
})

export const adminEventsAtom = atom<DogEvent[]>({
  key: 'adminEvents',
  default: [],
  effects: [logEffect, sessionStorageEffect, adminRemoteEventsEffect],
})

const EntryStartWeeks = 6
const EntryEndWeeks = 3

export const defaultEntryStartDate = (eventStartDate: Date) => sub(eventStartDate, { weeks: EntryStartWeeks })
export const defaultEntryEndDate = (eventStartDate: Date) => sub(eventStartDate, { weeks: EntryEndWeeks })

export const newEventStartDate = zonedStartOfDay(nextSaturday(addDays(Date.now(), 90)))
export const newEventEntryStartDate = defaultEntryStartDate(newEventStartDate)
export const newEventEntryEndDate = defaultEntryEndDate(newEventStartDate)

export const isDetaultEntryStartDate = (date: Date | undefined, eventStartDate: Date) =>
  !date || isSameDay(defaultEntryStartDate(eventStartDate), date)
export const isDetaultEntryEndDate = (date: Date | undefined, eventStartDate: Date) =>
  !date || isSameDay(defaultEntryEndDate(eventStartDate), date)

export const adminNewEventAtom = atom<DogEvent>({
  key: 'newEvent',
  default: {
    state: 'draft',
    startDate: newEventStartDate,
    endDate: newEventStartDate,
    entryStartDate: newEventEntryStartDate,
    entryEndDate: newEventEntryEndDate,
    classes: [],
    judges: [{ id: 0, name: '', official: true }],
  } as unknown as DogEvent,
  effects: [logEffect, localStorageEffect],
})

export const adminShowPastEventsAtom = atom<boolean>({
  key: 'adminShowPastEvents',
  default: false,
  effects: [logEffect, localStorageEffect],
})

export const adminEventFilterTextAtom = atom<string>({
  key: 'adminEventFilterText',
  default: '',
  effects: [logEffect, localStorageEffect],
})

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [logEffect, localStorageEffect],
})

export const adminEventClassAtom = atom<RegistrationClass | string>({
  key: 'adminEventClass',
  default: selector({
    key: 'adminEventClass/default',
    get: ({ get }) => {
      const event = get(adminCurrentEventSelector)
      return uniqueClasses(event)[0]
    },
  }),
  effects: [logEffect, sessionStorageEffect],
})

export const adminEventColumnsAtom = atom<GridColumnVisibilityModel>({
  key: 'adminEventColumns',
  default: { id: false },
  effects: [logEffect, localStorageEffect],
})

/**
 * Existing event editing, edits stored to session storage
 */
export const adminEditableEventByIdAtom = atomFamily<DogEvent | undefined, string>({
  key: 'adminEditableEvent/Id',
  default: adminEventSelector,
  effects: [logEffect, sessionStorageEffect],
})
