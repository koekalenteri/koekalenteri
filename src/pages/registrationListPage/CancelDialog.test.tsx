import { render, screen } from '@testing-library/react'
import { t } from 'i18next'

import { eventWithStaticDatesAndClass } from '../../__mockData__/events'
import { registrationWithStaticDatesAndClass } from '../../__mockData__/registrations'
import { sanitizeDogEvent } from '../../lib/event'

import { CancelDialog } from './CancelDialog'

const publicEventWithStaticDatesAndClass = sanitizeDogEvent(eventWithStaticDatesAndClass)

describe('CancelDialog', () => {
  it('should display secretary contact info when disabled', () => {
    const { baseElement } = render(
      <CancelDialog
        disabled
        event={publicEventWithStaticDatesAndClass}
        open
        registration={registrationWithStaticDatesAndClass}
        t={t}
      />
    )

    const { name = 'secretary name', phone = 'secretary phone' } =
      publicEventWithStaticDatesAndClass.contactInfo?.secretary ?? {}

    expect(
      screen.getByText(
        `Ilmoittautumisen voi perua kokeen alkamispäivänä ja kokeen alkamista edeltävänä päivänä vain ottamalla yhteyttä suoraan koesihteeriin ${name} ${phone}`
      )
    ).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })

  it('should render when not disabled', () => {
    const { baseElement } = render(
      <CancelDialog
        event={publicEventWithStaticDatesAndClass}
        open
        registration={registrationWithStaticDatesAndClass}
        t={t}
      />
    )

    expect(baseElement).toMatchSnapshot()
  })
})
