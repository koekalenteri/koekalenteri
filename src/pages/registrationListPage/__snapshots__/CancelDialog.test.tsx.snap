// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelDialog should display secretary contact info when disabled 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="cancel-dialog-description"
        aria-labelledby="cancel-dialog-title"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="cancel-dialog-title"
        >
          Ilmoittautumisen peruminen
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description"
          >
            Ilmoittautumisen voi perua koepäivänä vain ottamalla yhteyttä suoraan koesihteeriin Siiri Sihteeri 040-secretary
          </p>
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-1cajira-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description2"
          >
            Oletko varma, että haluat perua ilmoittautumisen?
          </p>
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-7dqnqj-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description3"
          >
            Katso tarkemmat peruutusehdot
             
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://yttmk.yhdistysavain.fi/noutajien-metsastyskokeet-2/ohjeistukset/kokeen-ja-tai-kilpailun-ilmoitta/"
              rel="noopener"
              target="_blank"
            >
              säännöistä
            </a>
            .
          </p>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Peru ilmoittautuminen
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Peruuta
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`CancelDialog should render when not disabled 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="cancel-dialog-description"
        aria-labelledby="cancel-dialog-title"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="cancel-dialog-title"
        >
          Ilmoittautumisen peruminen
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description"
          >
            Olet perumassa koiran TESTDOG-0030 Dog Name 20 ilmoittautumista kokeeseen 10.2.2021 NOME-B test location
          </p>
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-5mtc1h-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description2"
          >
            Oletko varma, että haluat perua ilmoittautumisen?
          </p>
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-7dqnqj-MuiTypography-root-MuiDialogContentText-root"
            id="cancel-dialog-description3"
          >
            Katso tarkemmat peruutusehdot
             
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://yttmk.yhdistysavain.fi/noutajien-metsastyskokeet-2/ohjeistukset/kokeen-ja-tai-kilpailun-ilmoitta/"
              rel="noopener"
              target="_blank"
            >
              säännöistä
            </a>
            .
          </p>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-focusVisible MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Peru ilmoittautuminen
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            >
              <span
                class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
              >
                <span
                  class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
                />
              </span>
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Peruuta
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
